### ISSUE TITLE
Samsung S9 - GPS popup coming twice

### ISSUE THREAD

**Issue by bajajsahil** [Baseflow/flutter-geolocator#544]
## 🐛 Bug Report

I am seeing the GPS popup coming twice when I press "Ok" for the first popup.

### Expected behavior
GPS popup should appear only once.

### Reproduction steps
This is happening just for the first time when I am opening the app with GPS turned off.

```dart
await getCurrentPosition()
```

### Configuration
[√] Flutter (Channel stable, 1.22.1, on Microsoft Windows [Version 10.0.17134.165], locale en-IN)
    • Flutter version 1.22.1 at D:\flutter\flutter
    • Framework revision f30b7f4db9 (2 days ago), 2020-10-08 10:06:30 -0700
    • Engine revision 75bef9f6c8
    • Dart version 2.10.1

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at D:\install\AndroidSDK
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = D:\install\AndroidSDK
    • Java binary at: D:\install\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Android Studio (version 4.0)
    • Android Studio at D:\install\Android Studio
    • Flutter plugin version 45.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] Connected device (1 available)
    • SM G960F (mobile) • 231173e32f027ece • android-arm64 • Android 10 (API 29)

• No issues found!

**Version:** 6.0.0+4

**Platform:** 
Android - Samsung S9

**Comment by bajajsahil** [comment]
hi @mvanbeusekom, did you get a chance to look at this issue?

**Comment by mvanbeusekom** [comment]
Hi @bajajsahil, no not yet but it is on the radar. I am a bit short pressed on time, but please feel free to do some initial debugging. It would really help if we could narrow down the problem to provide a fix. Especially if this is Samsung related, I only have a LG Nexus 5x device to test with and for now I haven't had this experience.

**Comment by bajajsahil** [comment]
Sure @mvanbeusekom. I Will update if I am able to find something.

**Comment by mvanbeusekom** [comment]
That is awesome thanks!

**Comment by manon-chaix** [comment]
Hi! I've the same issue .. Did you find a solution please ?

**Comment by bajajsahil** [comment]
> Hi! I've the same issue .. Did you find a solution please ?

Not yet. Didn't get a chance to look deep into this.

**Comment by florissmit1** [comment]
I reproduced the issue on a Galaxy S20 (Android 11). When requesting the current position in the example application, two pop-ups are showed. One to grant permission to use the GPS of the device, and one to turn on the GPS. I think this should be marked as a expected behaviour. Can u specify which pop-ups are showed on the device?

**Comment by mvanbeusekom** [comment]
That is indeed expected behavior and occurs if the user has disabled location services.

**Comment by bajajsahil** [comment]
@florissmit1 @mvanbeusekom Thanks for taking the time to look into this. 

I have specifically mentioned "GPS Popup (to turn on location service)" not permission popup. And, this is the reason I mentioned the device.

**Comment by JeroenWeener** [comment]
I am closing this issue as it has been over two years since the last comment.

If you are experiencing this issue and believe it is not expected behavior, please feel free to open a [new issue](https://github.com/Baseflow/flutter-geolocator/issues/new/choose), mentioning this one.
